mlir/lib/Dialect/Vector/Transforms/VectorTransforms.cpp

```diff
@@ -1607,7 +1607,24 @@ struct ChainedReduction final : OpRewritePattern<vector::ReductionOp> {
   }
 };
 
-/// For vectors with either leading or trailing unit dim, replaces:
+VectorType dropNonScalableUnitDimType(VectorType inVecTy) {
```

Comment:
Please document this hook. In particular, why only "NonScalableUnitDim"s are folded?

```suggestion
static VectorType dropNonScalableUnitDimType(VectorType inVecTy) {
```